src/waveform/renderers/allshader/digitsrenderer.cpp

```diff
@@ -131,12 +128,25 @@ void allshader::DigitsRenderer::updateTexture(
 
     m_height = static_cast<float>(std::ceil(maxTextHeight) + space * 2 + 1);
 
-    // Space around the digits
-    totalTextWidth += (space * 2 + 1) * NUM_CHARS;
-    totalTextWidth = std::ceil(totalTextWidth);
-
     const qreal y = maxTextHeight + space - 0.5;
 
+    qreal totalTextWidth{};
+    std::array<qreal, NUM_CHARS> xs;
+    for (int i = 0; i < NUM_CHARS; i++) {
+        const QString text(indexToChar(i));
+        xs[i] = totalTextWidth;
+        qreal w = std::round(metrics.horizontalAdvance(text) *
+                          devicePixelRatio / devicePixelRatio) +
+                space + space + 1.0;
```

Comment:
```suggestion
        qreal w = std::round(metrics.horizontalAdvance(text)) +
                space + space + 1.0;
```
Or should the division be after the round?

Reply:
Thanks for spotting. Wrongly placed parentheses.